Classes/Emogrifier.php

```diff
@@ -671,6 +671,7 @@ private function copyCssWithMediaToStyleNode(\DOMDocument $xmlDocument, \DOMXPat
             foreach ($this->parseSelectors($mediaQuery['css']) as $selector) {
                 if ($this->existsMatchForCssSelector($xpath, $selector['selector'])) {
                     $mediaQueriesRelevantForDocument[] = $mediaQuery['query'];
+                    break;
```

Comment:
Yes this is necessary to prevent adding the same mediaquery multiple times. If you look closely, you see that the loop walks through the selectors in the mediaquery. Any match in the document will add the whole mediaquery. If multiple selectors match, the old code (without the break) would add the complete mediaquery multiple times.

The break prevents this. At the first matching selector, the whole media query is added and the loop is aborted.